changelog.d/2258.bugfix.md

```diff
@@ -0,0 +1 @@
+Using `--upgrade-package` (and dynamically building project metadata) no longer causes an {exc}`AttributeError` when a pip internals change causes an error -- by {user}`Epic_Wink` and {user}`tusharsadhwani`.
```

Comment:
@EpicWink @sirosen looks like "causes an error" should be dropped from the phrase..

Reply:
Good catch. I'll submit a small follow up to correct it in just a bit.